### PR TITLE
Calculate loopSize based on post align adjusted size

### LINKED ISSUE #47094: Assertion failed 'paddingToAdd == paddingNeeded'
Job : [runtime-coreclr jitstressregs 20210117.1](https://dev.azure.com/dnceng/public/_build/results?buildId=954707&view=results)

Failed test : 
```
CoreCLR windows x64 Checked jitstressregs2 @ Windows.10.Amd64.Open
        - JIT\\HardwareIntrinsics\\General\\Vector256_1\\Vector256_1_ro\\Vector256_1_ro.cmd
        - JIT\\HardwareIntrinsics\\General\\Vector128_1\\Vector128_1_ro\\Vector128_1_ro.cmd
```
Part of error message:
```
Assert failure(PID 548 [0x00000224], Thread: 972 [0x03cc]): Assertion failed 'paddingToAdd == paddingNeeded' in 'JIT.HardwareIntrinsics.General.VectorGetAndWithLowerAndUpper__GetAndWithLowerAndUpperInt16:ValidateWithResult(System.Int16[],System.Int16[],System.String):this' during 'Emit code' (IL size 291)

    File: F:\workspace\_work\1\s\src\coreclr\jit\emitxarch.cpp Line: 9417
    Image: C:\h\w\AB90094A\p\CoreRun.exe


Return code:      1

 ...  ...

Beginning scenario: RunBasicScenario
Expected: 100
Actual: -1073740286
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=C:\h\w\AB90094A\p
> C:\h\w\AB90094A\w\AA5E0942\e\JIT\HardwareIntrinsics\General\Vector128_1\Vector128_1_ro\Vector128_1_ro.cmd

stack trace:
at JIT_HardwareIntrinsics._General_Vector256_1_Vector256_1_ro_Vector256_1_ro_._General_Vector256_1_Vector256_1_ro_Vector256_1_ro_cmd()
```

### PR DESCRIPTION
**Pull request by kunalspathak**
During outputting the `align` instruction, we recalculate the alignment needed to make sure that the padding we calculated earlier still matches when calculated on the basis of actual buffer address. In that, we calculate loopSize in `getLoopSize()`. While walking through the IGs in the loop, if we encounter that an IG is also marked as `IGF_LOOP_ALIGN`, we know that its size included the `align` instruction size and we were taking off that amount from `loopSize` (because that is not part of a loop). However, during outputting, we would have also adjusted the `align` instruction size of such IG and we should take that into account instead of max size of `align` instruction.

~I have added a field `paddingAdded` in `instrDescAlign` that tracks the adjusted padding amount which will be used to deduct it when we calculate `loopSize` during outputting the instruction. Since the check in `emitOutputAlign()` is under `#ifdef DEBUG`, I am including the `paddingAdded` field under `#ifdef DEBUG` as well. During release, we will never need this and hence will be unnecessary.~

I have also made minor changes in `JITDUMP`.

Fixes: https://github.com/dotnet/runtime/issues/47094

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib , @AndyAyersMS

File: src/coreclr/jit/emit.cpp [hunk 1]
```diff
@@ -4657,7 +4657,7 @@ bool emitter::emitEndsWithAlignInstr()
 //
 //  Returns:  size of a loop in bytes.
 //
-unsigned emitter::getLoopSize(insGroup* igLoopHeader, unsigned maxLoopSize)
+unsigned emitter::getLoopSize(insGroup* igLoopHeader, unsigned maxLoopSize DEBUG_ARG(bool isAlignAdjusted))
```
Comment (BruceForstall) [inline src/coreclr/jit/emit.cpp:4660]: The new arg needs to be documented above.

File: src/coreclr/jit/emit.cpp [hunk 2]
```diff
@@ -4896,8 +4945,7 @@ void emitter::emitLoopAlignAdjustments()
 //     3b. If the loop already fits in minimum alignmentBoundary blocks, then return 0. // already best aligned
 //     3c. return paddingNeeded.
 //
-unsigned emitter::emitCalculatePaddingForLoopAlignment(insGroup* ig,
-                                                       size_t offset DEBUG_ARG(bool displayAlignmentDetails))
+unsigned emitter::emitCalculatePaddingForLoopAlignment(insGroup* ig, size_t offset DEBUG_ARG(bool isAlignAdjusted))
```
Comment (BruceForstall) [inline src/coreclr/jit/emit.cpp:4948]: Please add a standard function header section describing the arguments, including the new argument.

**Review by BruceForstall — CHANGES_REQUESTED** [review]

File: src/coreclr/jit/emit.cpp (hunk 1 above)
Comment (kunalspathak) [reply src/coreclr/jit/emit.cpp:4663]: done.

**Comment by kunalspathak** [comment]
libraries-jitstressregs failures are fixed in https://github.com/dotnet/runtime/pull/47032.

**Review by BruceForstall — APPROVED** [review]

**Comment by kunalspathak** [comment]
`runtime-coreclr jitstressregs` failures are related to #11063.

**Pull request merged**